integration/container/logs_test.go

```diff
@@ -33,3 +35,135 @@ func TestLogsFollowTailEmpty(t *testing.T) {
 	_, err = stdcopy.StdCopy(ioutil.Discard, ioutil.Discard, logs)
 	assert.Check(t, err)
 }
+
+type daemonResources [2]int
+
+func (r daemonResources) String() string {
+	return fmt.Sprintf("goroutines: %d, file descriptors: %d", r[0], r[1])
+}
+
+func (r daemonResources) Delta(r2 daemonResources) (d daemonResources) {
+	for i := 0; i < len(r); i++ {
+		d[i] = r2[i] - r[i]
+		if d[i] < 0 { // negative values do not make sense here
+			d[i] = 0
+		}
+	}
+	return
+}
+
+// Test for #37391
+func TestLogsFollowGoroutineLeak(t *testing.T) {
```

Comment:
I feel like this can be a unit test... trying this out.

Reply:
Let me know what you think: https://gist.github.com/cpuguy83/e538793de18c762608358ee0eaddc197

Reply:
WDYT about this unit test?

Reply:
Thanks, from the first glance it makes sense; need some time to test and incorporate it.

Reply:
The only problem is this test can't be used before the commit that introduces Producer/Consumer and fixes the bug. Solved by taking the second test case and changing it to old-style `lw.Close()`.